packages/app-provider/src/ApolloContainer.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import * as R from 'ramda';
+import { ApolloProvider } from 'react-apollo';
+import { InMemoryCache, IntrospectionFragmentMatcher } from 'apollo-cache-inmemory';
+import { AuthClient, Authorizable, AuthState } from '@8base/utils';
+import { ApolloContainerPassedProps } from './types';
+import { FragmentsSchemaContainer } from './FragmentsSchemaContainer';
+
+const { EightBaseApolloClient } = require('@8base/apollo-client');
+const { withAuth } = require('@8base/auth');
+
+
+
+type ApolloContainerProps = ApolloContainerPassedProps & {
+  withAuth: boolean,
+  children: React.ReactNode,
+}
+
+type ApolloContainerHocProps = ApolloContainerProps & {
+  auth: AuthClient & Authorizable & {
+    authState: AuthState,
+  },
+}
+
+
+const ApolloContainer: React.ComponentType<ApolloContainerProps> = withAuth(
+  class ApolloContainer extends React.Component<ApolloContainerHocProps> {
+    client: any;
+
+    static defaultProps = {
+      auth: {},
+      withAuth: true,
+    }
+
+    onIdTokenExpired = async () => {
+      const {
+        auth: {
+          setAuthState,
+          renewToken,
+        },
+      } = this.props;
+
+      const { idToken } = await renewToken({});
+
+      await setAuthState({ token: idToken });
+    };
+
+    onAuthError = async () => {
+      const {
+        auth: {
+          purgeAuthState,
+          logout,
+        },
+      } = this.props;
+
+      await purgeAuthState();
+
+      if (typeof logout === 'function') {
+        await logout();
+      }
+    };
+
+    getAuthState = () => {
+      const {
+        auth: {
+          authState,
+        },
+      } = this.props;
+
+      return authState;
+    };
+
+    createClient = R.memoizeWith(R.identity, (fragmentsSchema) => {
```

Comment:
@ej9x also memoizeWith doesn't work with objects. It converts them as `[object Object]` for cache key.

Reply:
This is not my code:) I didn't change this part. It should work.

Reply:
I know. Just mentioned.

Reply:
What should I do?